src/core/settings.c

```diff
@@ -711,6 +743,31 @@ QuicSettingsLoad(
             &ValueLen);
         Settings->VersionNegotiationExtEnabled = !!Value;
     }
+
+    uint16_t MinimumMtu = Settings->MinimumMtu;
+    uint16_t MaximumMtu = Settings->MaximumMtu;
+    if (!Settings->IsSet.MinimumMtu) {
+        ValueLen = sizeof(MinimumMtu);
+        CxPlatStorageReadValue(
+            Storage,
+            QUIC_SETTING_MINIMUM_MTU,
+            (uint8_t*)&MinimumMtu,
+            &ValueLen);
+    }
+    if (!Settings->IsSet.MaximumMtu) {
+        ValueLen = sizeof(MaximumMtu);
+        CxPlatStorageReadValue(
+            Storage,
+            QUIC_SETTING_MAXIMUM_MTU,
+            (uint8_t*)&MaximumMtu,
+            &ValueLen);
+    }
+    if (MaximumMtu > MinimumMtu && MaximumMtu <= CXPLAT_MAX_MTU) {
+        Settings->MaximumMtu = MaximumMtu;
+    }
+    if (MinimumMtu >= QUIC_MIN_INITIAL_PACKET_LENGTH && MinimumMtu <= MaximumMtu) {
+        Settings->MinimumMtu = MinimumMtu;
+    }
```

Comment:
What if min > max?

Reply:
Then I don't want to set anything, just leave the defaults. We don't have any other sane option.

Reply:
Oh, I see what you're doing. You only accept the value if it's in a valid range. I would prefer you cap it to the valid range instead. So if MaximumMtu is > CXPLAT_MAX_MTU, use CXPLAT_MAX_MTU instead.

Reply:
For instance, like `MaxBytesPerKey` above.

Reply:
Makes sense. Done